test/testlib/minikube_utilities.go

```diff
@@ -52,27 +91,37 @@ func Teardown(name string) {
 
 /**
 * Verify all teardownLists have been executed already; and throw an ASSERT if not.
-* Can be used to verify correct codng of a test that uses teardown.
+* Can be used to verify correct coding of a test that uses teardown - and to ensure eventual release of resources.
 *
-* NOTE: while the funcs are called in the correct order for each list, there can be
-* NO guarantee that the lists are iterated in the correct order.
+* NOTE: while the funcs are called in the correct order for each list,
+* there can be NO guarantee that the lists are iterated in the correct order.
 *
 * This function MUST NOT be used as a replacement for calling teardown() at the correct point in the code.
  */
 func VerifyTeardown(t *testing.T) {
-	remaining := len(teardownLists)
+
+	// ensure all funcs in all lists are released
+	defer func() { teardownLists = make(map[string][]func()) }()
+	defer func() { diagnosticTeardownLists = make(map[string][]func()) }()
+
+	// append each diagnostic list to the corresponding (possibly empty) teardown list
+	for name, list := range diagnosticTeardownLists {
+		teardownLists[name] = append(teardownLists[name], list...)
+	}
+
+	// release all remaining resources - this is a "best effort" as the order of iterating the map is arbitrary
+	uncleared := make([]string, 0)
 
 	// make a "best-effort" at releasing all remaining resources
-	for _, list := range teardownLists {
+	for name, list := range teardownLists {
+		uncleared = append(uncleared, name)
+
 		for x := len(list) - 1; x >= 0; x-- {
 			list[x]()
 		}
 	}
 
-	// release all funcs in all lists
-	teardownLists = make(map[string][]func())
-
-	assert.True(t, remaining == 0, "Error - %d teardownLists were left uncleared", remaining)
+	assert.True(t, len(uncleared) == 0, "Error - %d teardownLists were left uncleared: %s", len(uncleared), uncleared)
```

Comment:
we now have `assert.Zero` or `assert.Equals` :)

Reply:
Good catch - thanks!
I thought I had changed them all.

Reply:
Actually - I realise that that assert.True() code is in master after the most recent updates which had changed many other of those to assert.Equal().
So I left it intentionally - assuming it had not been changed for a reason. :)

Reply:
I probably missed it

Reply:
I agree the error message is probably better with assert.Equal()...